### PR TITLE
VAR handling in LocalVariableDeclaration

### PR DESCRIPTION
**Pull request by DagSonntag**
The pull request for the issue discussed in issue #2924. I.e. that VAR is never detected as a keyword in current implementation of localvariabledeclaration.

Included examples with correct parsing

### CODE REVIEW

File: java/java/JavaParser.g4 [hunk 1]
```diff
@@ -254,7 +254,7 @@ arrayInitializer
     ;
 
 classOrInterfaceType
-    : typeIdentifier typeArguments? ('.' identifier typeArguments?)*
+    : (identifier typeArguments? '.')* typeIdentifier typeArguments?
```
Comment (kaby76) [inline java/java/JavaParser.g4:257]: Looks good. Annotations? https://docs.oracle.com/javase/specs/jls/se19/html/jls-4.html#jls-ClassOrInterfaceType

**Comment by teverett** [comment]
@DagSonntag  thanks!

File: java/java/JavaParser.g4 (hunk 1 above)
Comment (DagSonntag) [reply java/java/JavaParser.g4:257]: Hmm, this was closed before I could respond, but this is covered in the typearguments no?
Anyway, let's check it up when it comes up next time...

```
typeArguments
    : '<' typeArgument (',' typeArgument)* '>'
    ;

typeArgument
    : typeType
    | annotation* '?' ((EXTENDS | SUPER) typeType)?
    ;

typeType
    : annotation* (classOrInterfaceType | primitiveType) (annotation* '[' ']')*
    ;
```

**Comment by raghav-deepsource** [comment]
@DagSonntag , could you tell me how you generated the .tree file?

**Comment by kaby76** [comment]
> could you tell me how you generated the .tree file?

@raghav-deepsource I never liked how this is implemented in the tester. These files should be retained so that they can be renamed for remastering the test. For now, create an empty .tree file for each test input file you want to record, then run `mvn test`. In the output, edit the empty .tree files and add in the tree that mvn prints out. The .tree files should be one line, and no newline chars at the end of the file.

**Comment by raghav-deepsource** [comment]
hmm, yup, that worked. I was very confused when I saw that diff syntax the antlr test gave me. It would be better to directly print the parse result, so that can be more easily compared with the existing file (if the changed data were pasted into the original file, provided the file was checked into git already, there would be a nice git diff representation of the changes) instead of directly writing a diff out.

**Pull request merged**